BookPlayer/Models/Book+CoreDataClass.swift

```diff
@@ -106,10 +106,45 @@ public class Book: LibraryItem {
 
         let legacyIdentifier = bookUrl.originalUrl.lastPathComponent
         let storedTime = UserDefaults.standard.double(forKey: legacyIdentifier)
-        //migration of time
+
+        // migration of time
         if storedTime > 0 {
             self.currentTime = storedTime
             UserDefaults.standard.removeObject(forKey: legacyIdentifier)
         }
     }
+
+    override func getBookToPlay() -> Book? {
+        return self
+    }
+
+    func nextBook() -> Book? {
+        if
+            let playlist = self.playlist,
+            let next = playlist.getNextBook(after: self) {
```

Comment:
This is crashing with index out of range when it finishes the last item of a playlist